lib/bioinfo/blast.ml

```diff
@@ -0,0 +1,72 @@
+open Core
+open Bistro.Std
+open Bistro.EDSL
+open Bistro_bioinfo.Std
+
+type db = [`blast_db] directory
+let env = docker_image ~account:"pveber" ~name:"ncbi-blast" ~tag:"2.4.0" ()
+
+let db_name = "db"
+
+let fastadb fa dbtype = 
+	workflow ~descr:"blast.makedb" [
+		mkdir_p dest ; 
+		cmd ~env "makeblastdb" [
+			opt "-in" dep fa ;
+			opt "-dbtype" ident dbtype ; 
+			opt "-out" ident (dest // db_name) ;  	
+		] ; 	
+	]
+
+(* Basic blastn*)
+
+let blastn ?evalue ?word_size ?task ?gapopen ?gapextend ?penalty
+?reward ?outfmt ?perc_identity ?qcov_hsp_perc ?max_hsps ?max_target_seqs ?(threads = 4) db query out_name = (*See blastn documentation to know what options are*)
+	workflow ~descr:"blastn" ~np:threads [ 
+		mkdir_p dest ; 
+		cmd "blastn" ~env [
+			opt "-db" ident (dep db // db_name) ; 
+			opt "-query" dep query ; 
+			opt "-out" ident (dest // out_name) ; 
```

Comment:
what's the motivation for creating a directory here? Couldn't we give `dest` to the `-out` option?

Reply:
Because it's already created in makedb ? Or do we have to create it in the main pipeline ?

Reply:
Well yes, the same could be said about the database, I think in both cases you don't really need to create a directory and put the resulting file in it. The resulting file should be put at the location `dest` directly I think.

Reply:
To be more precise in think your wrapper could be simplified like this:
```ocaml
opt "-db" dep db ;
opt "-out" ident dest ;
```
of course the wrapper for `makedb` should be modified accordingly.